### PR TITLE
Unset certain environment variables before testing

### LINKED ISSUE #2657: make test should unset env variables such as KOPS_BASE_URL
running `make test` with certain variable set can cause failures.  We should unset all kops specific env variables before we test.

```
PROTOKUBE_IMAGE
NODEUP_URL
KOPS_BASE_URL
DNSCONTROLLER_IMAGE
KOPS_RUN_OBSOLETE_VERSION
ENV_VAR_CNI_VERSION_URL
```

To name a few. `os.Getenv` should be search for in the code base.

### PR DESCRIPTION
**Pull request by rifelpet**
Fixes #2657 and #4339

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -210,6 +210,7 @@ hooks: # Install Git hooks
 	cp hack/pre-commit.sh .git/hooks/pre-commit
 
 .PHONY: test
+unexport PROTOKUBE_IMAGE NODEUP_URL KOPS_BASE_URL DNSCONTROLLER_IMAGE KOPS_RUN_OBSOLETE_VERSION ENV_VAR_CNI_VERSION_URL AWS_REGION AWS_ACCESS_KEY_ID AWS_SECRET_ACCESS_KEY AWS_SESSION_TOKEN AWS_PROFILE
```
Comment (robinpercy) [inline Makefile]: What's the scope of unexport here? Will it only apply to the `test` recipe, or every recipe below it as well?

**Comment by robinpercy** [comment]
thanks @rifelpet I'll do some testing with this shortly.

File: Makefile (hunk 1 above)
Comment (robinpercy) [reply Makefile]: @rifelpet it appears that unexport has global scope, so we should put it at the top of the file for clarity.

**Comment by robinpercy** [comment]
Unexport causes all make targets to ignore the specified environment variables. I don't *think* any of those included in this PR are included at runtime. But would like a second pair of eyes from @chrislovecnm or @justinsb

**Comment by rifelpet** [comment]
@robinpercy Moved the unexport towards the top of the Makefile

**Comment by chrislovecnm** [comment]
@robinpercy I think we want to unexport all those variables and I think more.  As long as make is not crazy and unsets the variable in my current shell environment, I think we are good.

@rifelpet did you get all those pesky variables?  If I can read today, which is often debatable, I think we are missing STATE_STORE.  Another issue is to document all these pesky rabbits, aka ENV VARs.

**Comment by chrislovecnm** [comment]
Oh and we just merged KOPS_CLUSTER_NAME I think ...

File: Makefile [hunk 2]
```diff
@@ -45,6 +45,11 @@ API_OPTIONS?=
 # See http://stackoverflow.com/questions/18136918/how-to-get-current-relative-directory-of-your-makefile
 MAKEDIR:=$(strip $(shell dirname "$(realpath $(lastword $(MAKEFILE_LIST)))"))
 
+# Unexport environment variables that can affect tests and are not used in builds
```
Comment (robinpercy) [inline Makefile]: @rifelpet Nit: could we sort these in alphabetical order? It will make future management of the list less error-prone.

Comment (rifelpet) [reply Makefile]: Yeah, I have them categorized by kops-specific, aws-specific, and other cloud providers, but ill switch it to entirely alphabetical

**Comment by robinpercy** [comment]
Thanks @rifelpet!

@chrislovecnm were there any other variables you wanted to see in here?
/assign @chrislovecnm

**Pull request merged**